tool/common/common.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// GetPaginatedSessions grabs up to 'max' sessions.
+func GetPaginatedSessions(ctx context.Context, fromUTC, toUTC time.Time, pageSize int, order types.EventOrder, max int, authClient auth.ClientI) ([]events.AuditEvent, error) {
+	prevEventKey := ""
+	sessions := []events.AuditEvent{}
+	for {
+		nextEvents, eventKey, err := authClient.SearchSessionEvents(fromUTC, toUTC,
+			pageSize, order, prevEventKey, nil /* where condition */, "" /* session ID */)
```

Comment:
If the backend is dynamo and we don't specify a session ID, this will make a separate query for each day in the range, which can really hammer the backend with requests.

Should we enforce some maximum number of days you're allowed to search? If not, someone can specify a time of 0 and put a lot of stress on the system to make all of those queries.

Reply:
I can add a limit if you think that would make a better experience, but since the default range is one day in the past and to have more the user would need to explicitly ask for a wider range i'm not sure preventing them would be a very pleasant user experience. 
Maybe a warning would be better?

Reply:
I think we probably need some sort of upper limit.

The issue is if they request a very large range, Teleport will continue to make a query for each day, _even if they ctrl+c and abandon the request_.

Reply:
I have added in a limit of 365 days.